state.go

```diff
@@ -278,6 +297,9 @@ func (uc *UpdateCommitState) Handle(c Controller) (State, bool) {
 		// return NewUpdateErrorState(merr, uc.update), false
 	}
 
+	// cleanup state data
+	RemoveStateData(ctx.store)
```

Comment:
Are we sending some status back to server? I think it is important to make sure that communication with server happens before we are removing state data.